### PR TITLE
Feature/#156 list of recursive models

### PR DESCRIPTION
**Pull request by tsolbjor**
Keep track of the subAcceptors that have been applied and do not run multiple times for the same type.

#156

### CODE REVIEW

File: samples/Microsoft.Azure.WebJobs.Extensions.OpenApi.FunctionApp.Models/DummyResponseModel.cs [hunk 1]
```diff
@@ -54,5 +54,7 @@ public class DummyResponseModel
         public DummySubResponseModel SubResponse2 { get; set; }
 
         public DummyGenericModel<DummyModel> DummyGenericDummyModel { get; set; }
+
+        public List<DummyResponseModel> Children { get; set; }
```
Comment (justinyoo) [inline samples/Microsoft.Azure.WebJobs.Extensions.OpenApi.FunctionApp.Models/DummyResponseModel.cs:58]: Can we add a new `public Dictionary<string, DummyResponseModel> Children2 { get; set; }` property?

File: src/Microsoft.Azure.WebJobs.Extensions.OpenApi.Core/Visitors/ListObjectTypeVisitor.cs [hunk 2]
```diff
@@ -54,14 +56,21 @@ public override void Visit(IAcceptor acceptor, KeyValuePair<string, Type> type,
...
+                };
+                this.visitedTypes.Add(underlyingType, subAcceptor);
+                subAcceptor.Accept(this.VisitorCollection, namingStrategy);
 
-            subAcceptor.Accept(this.VisitorCollection, namingStrategy);
+            }
+            else
+            {
+                subAcceptor = this.visitedTypes[underlyingType];
+            }
```
Comment (justinyoo) [inline src/Microsoft.Azure.WebJobs.Extensions.OpenApi.Core/Visitors/ListObjectTypeVisitor.cs:73]: I'd like to ask you to add this logic to the `DictionaryObjectTypevisitor` as well.

File: test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp.csproj [hunk 3]
```diff
@@ -23,6 +23,7 @@
   </ItemGroup>
 
   <ItemGroup>
+    <ProjectReference Include="..\..\samples\Microsoft.Azure.WebJobs.Extensions.OpenApi.FunctionApp.Models\Microsoft.Azure.WebJobs.Extensions.OpenApi.FunctionApp.Models.csproj" />
```
Comment (justinyoo) [inline test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp.csproj:26]: In the integration test app, I'd like to be isolated as much as possible. Would you please remove this reference?

**Review by justinyoo — CHANGES_REQUESTED** [review]
@tsolbjor Thanks for your PR! I like the idea you implemented!

Could you please add the same logic where you put into `ListObjectTypeVisotor` to `DictionaryObjectTypeVisotor` so that both array types and dictionary types work in the same way as each other?

File: test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp/Get_ApplicationJson_GenericAndRecursive_HttpTrigger.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Attributes;
+using Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp.Models;
+using Microsoft.Extensions.Logging;
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp
+{
+    public static class Get_ApplicationJson_GenericAndRecursive_HttpTrigger
```
Comment (tsolbjor) [inline test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp/Get_ApplicationJson_GenericAndRecursive_HttpTrigger.cs:13]: Couldnt find any unittest that failed with these objecttypes - therefore created a new variant of the object in the TestApp

Comment (justinyoo) [reply test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp/Get_ApplicationJson_GenericAndRecursive_HttpTrigger.cs:13]: Thanks for the addition! I'll take further look for its corresponding integration tests.

**Review by justinyoo — APPROVED** [review]
@tsolbjor Thanks for your PR! It LGTM!

**Pull request merged**